### ISSUE TITLE
Ability to "Go To Definition" while writing code

### ISSUE THREAD

**Issue by TysonMN** [elmish/Elmish.WPF#311]
This is a spin off of https://github.com/fslaborg/flips/issues/66

The development experience for consumers of Elmish.WPF would be improved if it were possible to "Go To Definition" for things defined in Elmish.WPF when using Elmish.WPF.  As discussed in that issue, [SourceLink](https://github.com/dotnet/sourcelink) might make that possible.  I briefly tried but failed to get that functionality.

Elmish.WPF briefly included SourceLink (added in PR #129) and it was removed in favor of embedding the source in the NuGet package (see issue #196).

I know that embedding the source in the NuGet package allows one to step into Elmish.WPF functions when debugging but does not allow one to "Go To Definition" for the same function when not debugging (even if the source file remains open after stepping into it while debugging).

It is possible that the inability to "Go To Definition" when the source is included in the NuGet package is a bug in Visual Studio (and maybe specific to F#).  If so, I will submit a bug report, but I want to both be confident that I understand what doesn't work and have minimal reproductions provide.  I want to test different IDEs (Visual Studio vs others like Rider or Ionide in VS Code) and the consuming code being C# or F#.

The SourceLink README says the disadvantage of including the source in the NuGet package is larger NuGet package sizes, which causes longer restore times.  (It would also consume more disk space when the application is installed/deployed, but they didn't mention that.  Another disadvantage is when the source code should remain a secret, which is not the case for an open source project, but they didn't mention that disadvantage either.)  This disadvantage of larger NuGet package sizes doesn't bother me in the slightest.  In my testing just now, the NuGet package changes from ~350 KB to ~450 KB when including the source.  I am not going to add a layer of indirection (i.e. fetching the source from GitHub instead of including it in the NuGet package) to save ~100 KB.

**Comment by TysonMN** [comment]
Trying following [these steps](https://aaronstannard.com/visual-studio-sourcelink-setup/) (c.f. https://github.com/fslaborg/flips/issues/66#issuecomment-747455833).

**Comment by cmeeren** [comment]
A couple thoughts:

* Yesterday I launched [Facil](https://github.com/cmeeren/Facil), and I tried to follow [the steps recently posted on the .NET blog](https://devblogs.microsoft.com/dotnet/producing-packages-with-source-link/). No matter what I tried, I could not get the produced package to go "green" in the SourceLink health status in NuGet Package Explorer. Embedding the sources (just like we do here) was the only way I was able to do that. Maybe I am doing something wrong (though the instructions are fairly simple), or maybe NuGet Package Explorer is bugged (I found it impossible to go green on the "compiler flags" and "deterministic build" statuses, too), or maybe the packaging in Facil is just too complicated (several binaries and a build tool are placed in the same package).
* If you want to investigate and you find a way to make it work, feel free to open a PR.
* For VS, this seems to be tracked in https://github.com/dotnet/roslyn/issues/24349. Not sure if there's a difference between C# and F#.
* I may be mistaken, but it seems that the expected functionality should be achievable regardless of whether the sources are remote or embedded. Perhaps a better (or additional) course of action would be to open an Ionide issue asking them to support embedded sources, too.

> This disadvantage of larger NuGet package sizes doesn't bother me in the slightest. In my testing just now, the NuGet package changes from ~350 KB to ~450 KB when including the source. I am not going to add a layer of indirection (i.e. fetching the source from GitHub instead of including it in the NuGet package) to save ~100 KB.

Agreed.
